tensorbay/opendataset/RarePlanesSynthetic/loader.py

```diff
@@ -0,0 +1,141 @@
+        path: The root directory of the dataset.
+
+    Returns:
+        Loaded :class:`~tensorbay.dataset.dataset.Dataset` instance.
+
+    """
+    root_path = os.path.abspath(os.path.expanduser(path))
+    dataset = Dataset(DATASET_NAME)
+    dataset.load_catalog(os.path.join(os.path.dirname(__file__), "catalog.json"))
+    category_ids = {
+        category: index
+        for index, category in enumerate(dataset.catalog.panoptic_mask.categories.keys())
+    }
+    segment = dataset.create_segment()
+    mask_dir = os.path.join(root_path, "masks")
+    annotation_dir = os.path.join(root_path, "xmls")
+    for image_path in glob(os.path.join(root_path, "images", "*")):
+        segment.append(_get_data(image_path, mask_dir, annotation_dir, category_ids))
+    return dataset
+
+
+def _get_data(
+    image_path: str, mask_dir: str, annotation_dir: str, category_ids: Dict[str, int]
+) -> Data:
+    stem = os.path.splitext(os.path.basename(image_path))[0]
+    mask_path = os.path.join(mask_dir, f"{stem}.png")
+    data = Data(image_path)
+    label = data.label
+    original_mask = np.array(Image.open(os.path.join(mask_dir, f"{stem}_mask.png")))
+    mask = np.array(np.zeros(original_mask.shape[:2], dtype=np.uint8))
+    with open(os.path.join(annotation_dir, f"{stem}.xml"), "r", encoding="utf-8") as fp:
+        labels = xmltodict.parse(fp.read())["image"]
+    all_category_ids: Dict[int, int] = {}
+    label.box2d = _get_box2d_and_modify_mask(
+        labels, mask, original_mask, category_ids, all_category_ids
+    )
+    Image.fromarray(mask).save(mask_path)
+    label.classification = _get_classification(labels["JSON_Variation_Parameters"]["parameter"])
+    panoptic_mask = PanopticMask(mask_path)
+    panoptic_mask.all_category_ids = all_category_ids
+    label.panoptic_mask = panoptic_mask
+    return data
+
+
+def _get_classification(classification_labels: List[Dict[str, str]]) -> Classification:
+    classification_attributes: Dict[str, Union[int, float, str]] = {}
+    for indexes, attribute_getter in _ATTRIBUTES_GETTER.items():
+        for index in indexes:
+            attribute = classification_labels[index]
+            classification_attributes[attribute["@name"]] = attribute_getter(attribute["@value"])
+    return Classification(attributes=classification_attributes)
+
+
+def _get_box2d_and_modify_mask(
+    labels: Dict[str, Any],
+    mask: np.ndarray,
+    original_mask: np.ndarray,
+    category_ids: Dict[str, int],
+    all_category_ids: Dict[int, int],
+) -> List[LabeledBox2D]:
+    objects = labels["object"]
+    box2ds: List[LabeledBox2D] = []
+    if not isinstance(objects, list):
+        objects = [objects]
+    for index, obj in enumerate(objects, 1):
+        category_parts = [obj["category0"]]
+        for i in range(1, 6):
+            category_part = obj.get(f"category{i}", "")
+            if category_part:
+                category_parts.append(category_part)
+        category = ".".join(category_parts)
```

Comment:
Is it possible to use functions in `itertools`, like `takewhile`?
`category = ".".join(takewhile(lambda x: x, (obj.get(f"category{i}", "") for i in range(0, 6))))`